editor/src/components/canvas/controls/select-mode/flex-gap-control.tsx

```diff
@@ -0,0 +1,201 @@
+  useHoverWithDelay,
+} from './control-common'
+
+interface FlexGapControlProps {
+  selectedElement: ElementPath
+  flexDirection: SimpleFlexDirection
+  updatedGapValue: CSSNumberWithRenderedValue
+}
+
+export const FlexGapControlTestId = 'FlexGapControlTestId'
+export const FlexGapControlHandleTestId = 'FlexGapControlHandleTestId'
+
+const PaddingIndicatorOffset = 10
+const HitAreaPadding = 5
+
+export const FlexGapControl = controlForStrategyMemoized<FlexGapControlProps>((props) => {
+  const { selectedElement, flexDirection, updatedGapValue } = props
+  const colorTheme = useColorTheme()
+  const indicatorColor = colorTheme.brandNeonPink.value
+
+  const [indicatorShown, setIndicatorShown] = useState<string | null>(null)
+  const [backgroundShown, setBackgroundShown] = useState<boolean>(false)
+
+  const [controlHoverStart, controlHoverEnd] = useHoverWithDelay(0, setBackgroundShown)
+
+  const handleHoverStart = React.useCallback((id: string) => setIndicatorShown(id), [])
+  const handleHoverEnd = React.useCallback(() => setIndicatorShown(null), [])
+
+  const { dispatch, scale, metadata, isDragging } = useEditorState(
+    (store) => ({
+      dispatch: store.dispatch,
+      scale: store.editor.canvas.scale,
+      metadata: store.editor.canvas.interactionSession?.latestMetadata ?? store.editor.jsxMetadata,
+      isDragging: store.editor.canvas.interactionSession?.activeControl.type === 'FLEX_GAP_HANDLE',
+    }),
+    'FlexGapControl dispatch scale',
+  )
+
+  const canvasOffset = useRefEditorState((store) => store.editor.canvas.roundedCanvasOffset)
+
+  const controlRef = useBoundingBox([selectedElement], (ref, boundingBox) => {
+    if (isZeroSizedElement(boundingBox)) {
+      ref.current.style.display = 'none'
+    } else {
+      ref.current.style.display = 'block'
+      ref.current.style.left = boundingBox.x + 'px'
+      ref.current.style.top = boundingBox.y + 'px'
+      ref.current.style.width = boundingBox.width + 'px'
+      ref.current.style.height = boundingBox.height + 'px'
+    }
+  })
+
+  const controlBounds = gapControlBoundsFromMetadata(
+    metadata,
+    selectedElement,
+    updatedGapValue.renderedValuePx,
+    flexDirection,
+  )
+
+  const onMouseDown = React.useCallback(
+    (e: React.MouseEvent<HTMLDivElement>) => {
+      startInteraction(e, dispatch, canvasOffset.current, scale)
+    },
+    [canvasOffset, dispatch, scale],
+  )
+
+  const { width, height } = handleDimensions(flexDirection, scale)
+
+  const [paddingIndicatorOffset, hitAreaPadding, dragBorderWidth, borderWidth] = [
+    PaddingIndicatorOffset,
+    HitAreaPadding,
+    1,
+    0.5,
+  ].map((v) => v / scale)
+
+  const shouldShowBackground = !isDragging && backgroundShown
+  const shouldShowIndicator = (path: string) => !isDragging && indicatorShown === path
+
+  return (
+    <CanvasOffsetWrapper>
+      <div data-testid={FlexGapControlTestId} ref={controlRef}>
+        {controlBounds.map(({ bounds, path: p }) => {
```

Comment:
I think everything inside this `map` could be extracted into another component, especially as this is creating an inline function for one of the event handlers below.